### DISCUSSION TITLE
REST API request duration

### DISCUSSION THREAD

**Discussion by emizzle** [codex-storage/codex-research#59] [Ideas]
There is currently a timeout set for a [pending block retrieval](https://github.com/status-im/nim-dagger/blob/26ead9726d755cfa1ec39f52a3ed6d0698df3416/dagger/blockexchange/engine.nim#L123) to 5 minutes. A consumer of the REST api may have to wait 5 minutes before getting a response for `/download`, which could be a block or a timeout. When the DHT is implemented, this may take even longer. Is waiting 5 minutes for a response acceptable? Should we consider alternative approaches, such as websockets?

**Comment by michaelsbradleyjr** [comment]
Another thing to consider is that the default http request timeout on the client-side varies by client. For some clients that timeout is configurable, but for some it's not.
